deepspeed/module_inject/replace_module.py

```diff
@@ -271,10 +272,22 @@ def replace_wo_policy(module, all_reduce_linears, prefix="", state_dict=None):
         # 2. Set the tensor parallelism config
         _autotp.set_tensor_parallel_config(config.tensor_parallel.tp_size, config.tensor_parallel.tp_group)
 
-        # 3. Set linear policies
+        # 3. Try to get num_key_heads from model_config.num_key_value_heads
+        num_kv_heads = None
+        kv_head_names = ['num_key_value_heads', 'num_attention_heads', 'n_heads']
+        for name in kv_head_names:
+            if hasattr(model_config, name):
+                num_kv_heads = getattr(model_config, name)
+                if num_kv_heads != None:
+                    break
+
+        # 5. When we have num_kv_heads defined, uneven division is possible, otherwise enforce even division
+        set_num_kv_heads(num_kv_heads)
+
+        # 6. Set linear policies
```

Comment:
Could we refactor this to follow the code structure we have established in this function? That is, each step is just a single line of code that calls to a function that has the all the code for completing that step.

Reply:
Also please fix the numbering. In the current state `# 4` is skipped.

Reply:
Updated, now each step is a single function call and numbering fixed.